openpype/tools/loader/model.py

```diff
@@ -614,6 +668,37 @@ def _fill_subset_items(
                 "index": self.index(group_item.row(), 0)
             }
 
+        def _add_subset_item(subset_doc, parent_item, parent_index):
+            last_version = last_versions_by_subset_id.get(
+                subset_doc["_id"]
+            )
+            # do not show subset without version
+            if not last_version:
+                return
+
+            data = copy.deepcopy(subset_doc)
+            data["subset"] = subset_doc["name"]
+
+            asset_id = subset_doc["parent"]
+            data["asset"] = asset_docs_by_id[asset_id]["name"]
+
+            data["last_version"] = last_version
+
+            loaded = subset_doc["_id"] in subsets_loaded_by_id
+            data["loaded_in_scene"] = "yes" if loaded else "no"
+
+            # Sync server data
+            data.update(
+                self._get_last_repre_info(repre_info_by_version_id,
+                                          last_version["_id"]))
+
+            item = Item()
+            item.update(data)
+            self.add_child(item, parent_item)
+
+            index = self.index(item.row(), 0, parent_index)
+            self.set_version(index, last_version)
```

Comment:
This was solely moved to a function inside the method to avoid the code duplication just a bit further down.